### PR TITLE
Enable reproducible builds

### PR DESCRIPTION
**Pull request by symbiogenesis**
[Reproducible builds](https://github.com/dotnet/reproducible-builds) are an important way to improve the integrity of binary releases.

### CODE REVIEW

File: PowerShell.Common.props [hunk 1]
```diff
@@ -192,4 +195,11 @@
   <PropertyGroup>
     <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
   </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="DotNet.ReproducibleBuilds" Version="1.1.1">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
```
Comment (TravisEz13) [inline PowerShell.Common.props:202]:
```suggestion
```

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
Summary: our build doesn't work with these settings even if the CI passes.  Please file a feature request as we would need to do this work.

Some of the things you could do to move towards this:

- Please add the following as described here, so that we always link to the public repo, https://devblogs.microsoft.com/nuget/introducing-source-code-link-for-nuget-packages/

```
    <RepositoryUrl>https://github.com/powershell/powershell.git</RepositoryUrl> 
    <RepositoryType>git</RepositoryType>
```   
- After that, you should be able to set `PublishRepositoryUrl` to `true`

- Set `EmbedUntrackedSources` to `true` this should be safe.

- Set `ContinuousIntegrationBuild` properly

```
<PropertyGroup Condition="'$(OB_OUTPUTDIRECTORY)' != ''">
  <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
</PropertyGroup>
```

Notes:
- `DebugType` cannot be changed except for the Debug configuration
- I don't think we have any WPF, so, I don't think `IncludePackageReferencesDuringMarkupCompilation` is relevant.
- Some parts of the build are not reproducible:
  -  Adding modules
  - ResGen is not run as part of the build:  https://github.com/PowerShell/PowerShell/blob/8a8f2d4b234221c8ea2c47902c75d7db20fb5c33/build.psm1#L2487
  - TypeGen is not part of the build: https://github.com/PowerShell/PowerShell/blob/8a8f2d4b234221c8ea2c47902c75d7db20fb5c33/build.psm1#L2438
  - This analysis is for the main packages.

**Comment by iSazonov** [comment]
> * Please add the following as described here, so that we always link to the public repo, https://devblogs.microsoft.com/nuget/introducing-source-code-link-for-nuget-packages/
> 
> ```
>     <RepositoryUrl>https://github.com/powershell/powershell.git</RepositoryUrl> 
>     <RepositoryType>git</RepositoryType>
> ```

Do we still need this? https://github.com/dotnet/sourcelink?tab=readme-ov-file#using-source-link-in-net-projects
> Starting with .NET 8, Source Link for the following source control providers is included in the .NET SDK and enabled by default:

>[GitHub](http://github.com/) or [GitHub Enterprise](https://enterprise.github.com/home)
[Azure Repos](https://azure.microsoft.com/en-us/services/devops/repos) git repositories (formerly known as Visual Studio Team Services)
[GitLab](https://gitlab.com/) 12.0+ (for older versions see [GitLab settings](https://github.com/dotnet/sourcelink?tab=readme-ov-file#gitlab))
[Bitbucket](https://bitbucket.org/) 4.7+ (for older versions see [Bitbucket settings](https://github.com/dotnet/sourcelink?tab=readme-ov-file#bitbucket))
If your project uses .NET SDK 8+ and is hosted by the above providers it does not need to reference any Source Link packages or set any build properties.

**Pull request closed without merging**